### ISSUE TITLE
Nodes ready before `aws-cni` is ready to assign IPs

### ISSUE THREAD

**Issue by alex-dabija** [giantswarm/roadmap#850]
## Story

_*- As a cluster admin, I want the nodes to be marked as ready only after `aws-cni` is ready and able to assign IPs to scheduled pods in order to have faster pod scheduling times and to have fewer pod failures on new nodes.*_

## Background

At the moment, a Kubernetes node is marked as ready as soon as it passes the standard `kubelet` checks. Giant Swarm AWS clusters have `aws-cni` running and it's slow to start and allocates IPs slowly to new pods. Also, `aws-cni` is not part of the standard Kubernetes checks to mark a node as ready. This leads to pods being scheduled to a node before `aws-cni` is ready to allocate IPs, causing customer pods to fail.

This issue is exacerbated on large clusters with lots of autoscaling events:
- CI/CD clusters;
- clusters with batch jobs;

## Changes

A new node starts with a taint set so that no customer pods are scheduled to the node. Another process is going to monitor if `aws-cni` is ready. When it become ready the taint is removed and pods can be scheduled on the node. 

## Questions:

- [ ] How are EKS clusters configured? Do they also have this problem? (This question applies only for managed nodes).

**Comment by alex-dabija** [comment]
Upstream issue: https://github.com/aws/amazon-vpc-cni-k8s/issues/1523 .

**Comment by AverageMarcus** [comment]
Differences between our deployed aws-node and the default for EKS:

* We set `AWS_VPC_K8S_CNI_RANDOMIZESNAT` to `none` where EKS uses the default of `prng`  (Docs: https://github.com/aws/amazon-vpc-cni-k8s#aws_vpc_k8s_cni_randomizesnat)
* We enabled `AWS_VPC_K8S_CNI_CUSTOM_NETWORK_CFG`
* We set `WARM_IP_TARGET` to 10 and `MINIMUM_IP_TARGET` to 40 (these are left unset in EKS)
	From the aws-node docs:
	> To be clear, only set WARM_IP_TARGET for small clusters, or clusters with very low pod churn.

The only other differences are either explicit setting of values matching defaults or related to logging.

**Comment by whites11** [comment]
Random observations:

- nodes do NOT become ready until `aws-node` Pod becomes ready.
- nodes DO become ready even before `calico-node` Pod becomes ready.

**Comment by AverageMarcus** [comment]
> nodes do NOT become ready until aws-node Pod becomes ready.
Do they become ready _because_ aws-node becomes ready? I wasn't sure if it was just a timing thing and just happened to become ready around the same time.

**Comment by whites11** [comment]
> > nodes do NOT become ready until aws-node Pod becomes ready.
> 
> Do they become ready _because_ aws-node becomes ready? I wasn't sure if it was just a timing thing and just happened to become ready around the same time.

sometime it takes minutes for aws-node to be Ready and only then the node becomes ready. I'm 99.9% sure it is a consequence

**Comment by alex-dabija** [comment]
We are moving to Cilium and we'll be dropping `aws-cni`.

**Comment by alex-dabija** [comment]
Old issue, `aws-cni` was dropped.
